internal/component/postgresql/metadata.go

```diff
@@ -63,32 +62,20 @@ func (m *postgresMetadataStruct) InitWithMetadata(meta state.Metadata) error {
 	}
 
 	// Timeout
-	s, ok := meta.Properties[timeoutKey]
-	if ok && s != "" {
-		timeoutInSec, err := strconv.ParseInt(s, 10, 0)
-		if err != nil {
-			return fmt.Errorf("invalid value for '%s': %s", timeoutKey, s)
-		}
-		if timeoutInSec < 1 {
-			return fmt.Errorf("invalid value for '%s': must be greater than 0", timeoutKey)
-		}
-
-		m.timeout = time.Duration(timeoutInSec) * time.Second
+	if m.Timeout < 1*time.Second {
+		return fmt.Errorf("invalid value for '%s': must be greater than 0", timeoutKey)
 	}
 
 	// Cleanup interval
-	s, ok = meta.Properties[cleanupIntervalKey]
-	if ok && s != "" {
-		cleanupIntervalInSec, err := strconv.ParseInt(s, 10, 0)
-		if err != nil {
-			return fmt.Errorf("invalid value for '%s': %s", cleanupIntervalKey, s)
-		}
-
+	if m.CleanupInterval != nil {
 		// Non-positive value from meta means disable auto cleanup.
-		if cleanupIntervalInSec > 0 {
-			m.cleanupInterval = ptr.Of(time.Duration(cleanupIntervalInSec) * time.Second)
-		} else {
-			m.cleanupInterval = nil
+		if *m.CleanupInterval <= 0 {
+			if meta.Properties[cleanupIntervalKey] == "" {
+				// unfortunately the mapstructure decoder decodes an empty string to 0, a missing key would be nil however
```

Comment:
Is there a way we can change this behavior so we don't need to go back to check if the string is empty? Such a behavior seems dangerous.

Reply:
No we cannot - I tried for a few hours, but this is not a feature of the library today unfortunately.

Reply:
I don't think it's a real issue - the test case of empty string is unrealistic. That's different from omitting the field.